### PR TITLE
Make killmessages support hex color

### PR DESCRIPTION
**Pull request by AboShanab**
Hello, In that pull request will make killmessages more modern, It will support hex color by Default

Best regards
AboShanab.

### CODE REVIEW

**Comment by Dutchman101** [comment]
I doubt about the current patch because it makes some colorcoded names look ugly-kind of cursive partially; see this screenshot.. 
colorc

Apart from that it's a great feature

File: [gameplay]/killmessages/utils/textlib.lua [hunk 1]
```diff
@@ -123,7 +124,12 @@ function dxText:destroy()
 end
 
 function dxText:extent()
-	local extent = dxGetTextWidth ( self.strText, self.fScale, self.strFont )
+	if ( self.bColorCoded == true ) then
```
Comment (qaisjp) [inline [gameplay]/killmessages/utils/textlib.lua]: `if self.bColorCoded then` is sufficient.

File: [gameplay]/killmessages/utils/textlib.lua [hunk 2]
```diff
@@ -123,7 +124,12 @@ function dxText:destroy()
 end
 
 function dxText:extent()
-	local extent = dxGetTextWidth ( self.strText, self.fScale, self.strFont )
+	if ( self.bColorCoded == true ) then
+		text = string.gsub ( self.strText, "#%x%x%x%x%x%x", "" )
```
Comment (qaisjp) [inline [gameplay]/killmessages/utils/textlib.lua]: Follow this format: `fname(arg, art, art)`

File: [gameplay]/killmessages/utils/textlib.lua [hunk 3]
```diff
@@ -123,7 +124,12 @@ function dxText:destroy()
 end
 
 function dxText:extent()
-	local extent = dxGetTextWidth ( self.strText, self.fScale, self.strFont )
+	if ( self.bColorCoded == true ) then
+		text = string.gsub ( self.strText, "#%x%x%x%x%x%x", "" )
+	else
+		text = self.strText
+	end
+	local extent = dxGetTextWidth ( text, self.fScale, self.strFont )
```
Comment (qaisjp) [inline [gameplay]/killmessages/utils/textlib.lua]: Follow this format: `fname(arg, arg, arg)`

File: [gameplay]/killmessages/utils/textlib.lua [hunk 4]
```diff
@@ -256,7 +262,7 @@ addEventHandler ( "onClientRender", getRootElement(),
 						for offsetX=-outlinesize,outlinesize,outlinesize do
 							for offsetY=-outlinesize,outlinesize,outlinesize do
 								if not (offsetX == 0 and offsetY == 0) then
-									dxDrawText(self.strText, l + offsetX, t + offsetY, r + offsetX, b + offsetY, tocolor(att2, att3, att4, att5), self.fScale, self.strFont, self.bHorizontalAlign, self.bVerticalAlign, self.bClip, self.bWordWrap, self.bPostGUI )
+									dxDrawText(self.strText, l + offsetX, t + offsetY, r + offsetX, b + offsetY, tocolor(att2, att3, att4, att5), self.fScale, self.strFont, self.bHorizontalAlign, self.bVerticalAlign, self.bClip, self.bWordWrap, self.bPostGUI, self.bColorCoded )
```
Comment (qaisjp) [inline [gameplay]/killmessages/utils/textlib.lua]: Please fix formatting at end of life, even though it was there before

**Comment by AboShanab** [comment]
@qaisjp I fixed the formatting.
@Dutchman101, That's true.. do you have any suggest for fix that problem? maybe add custom shadow for player name that have hex color?

**Comment by AboShanab** [comment]
@Dutchman101, You can test it now, It's more modern and fixed that problem.

**Comment by Dutchman101** [comment]
Tested thorougly and stable, same opinion as qaisjp, so merging.

**Pull request merged**